test/e2e/network/service.go

```diff
@@ -1123,6 +1227,48 @@ var _ = SIGDescribe("Services", func() {
 		framework.ExpectNoError(verifyServeHostnameServiceUp(cs, ns, host, podNames2, svc2IP, servicePort))
 	})
 
+	/*
+	   Release : v1.19
+	   Testname: Service, preserve UDP traffic when pod backing a service cycles
+	   Description: Create a UDP ClusterIP Service with a backing server pod. Run traffic
+	   through the service to the server pod. The traffic flow must recover when the server
+	   pod originally serving requests is deleted and a new pod is ready to serve requests
+	*/
+	framework.ConformanceIt("should be able to preserve UDP traffic when server pod cycles [Slow]", func() {
+		serviceName := "clusterip-test"
+		serverPod1Name := "server-1"
+		serverPod2Name := "server-2"
+
+		ns := f.Namespace.Name
+
+		udpJig := e2eservice.NewTestJig(cs, ns, serviceName)
+		udpService, err := udpJig.CreateUDPService(nil)
+		framework.ExpectNoError(err)
+
+		serverPod1 := newAgnhostPod(serverPod1Name, "netexec", fmt.Sprintf("--udp-port=%d", 80))
+		serverPod1.Labels = udpJig.Labels
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), serverPod1, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, serverPod1.Name, f.Namespace.Name, framework.PodStartTimeout))
+
+		errorChannel := make(chan error)
+		go continuousEcho(udpService.Spec.ClusterIP, 80, 3*time.Second, 20, errorChannel)
```

Comment:
This assumes that the e2e.test binary has access to the clusterIP, right?

Reply:
@aojea  Yes, is that a bad assumption?

Reply:
I don't know if it is documented, but it doesn't seem a good idea if the e2e framework has to require this, I think that is not common that the test runner has access to the pod and clusterIP network.
I think that the best approach is to spawn another pod with the client `podClient` and execute it from the pod.
This e2e network performance test does something similar https://github.com/kubernetes/kubernetes/blob/2c2e9cdd72f363e6e6282f1ec2832fb6c2f4858a/test/e2e/network/networking_perf.go#L57-L61

You should verify this, but the agnost host has a `iperf` inside and maybe running in UDP mode can be valid for your test